lib/graphql/pagination/relation_connection.rb

```diff
@@ -47,7 +47,7 @@ def has_next_page
       def cursor_for(item)
         load_nodes
         # index in nodes + existing offset + 1 (because it's offset, not index)
-        offset = nodes.index(item) + 1 + (@paged_nodes_offset || 0) + (relation_offset(items) || 0)
+        offset = nodes.index(item) + 1 + (@paged_nodes_offset || 0) - (relation_offset(items) || 0)
```

Comment:
It seems like I just ported this wrong, here's the previous implementation: 

https://github.com/rmosolgo/graphql-ruby/blob/1db571aa7bd2ec253053ab970cd167b42b13aa3b/lib/graphql/relay/relation_connection.rb#L14